frontend/ui/presets.lua

```diff
@@ -0,0 +1,329 @@
+                                presets[preset_name] = preset_obj.buildPreset()
+                                UIManager:show(InfoMessage:new{
+                                    text = T(_("Preset '%1' was updated with current settings"), preset_name),
+                                    timeout = 2,
+                                })
+                            end,
+                        })
+                    end,
+                    other_buttons_first = true,
+                    other_buttons = {
+                        {
+                            {
+                                text = _("Delete"),
+                                callback = function()
+                                    UIManager:show(ConfirmBox:new{
+                                        text = T(_("Are you sure you want to delete preset '%1'?"), preset_name),
+                                        ok_text = _("Delete"),
+                                        ok_callback = function()
+                                            presets[preset_name] = nil
+                                            local action_key = preset_obj.dispatcher_name
+                                            if action_key then
+                                                UIManager:broadcastEvent(Event:new("DispatcherActionValueChanged", {
+                                                    name = action_key,
+                                                    old_value = preset_name,
+                                                    new_value = nil -- delete the action
+                                                }))
+                                            end
+                                            table.remove(touchmenu_instance.item_table, item.idx)
+                                            touchmenu_instance:updateItems()
+                                        end,
+                                    })
+                                end,
+                            },
+                            {
+                                text = _("Rename"),
+                                callback = function()
+                                    Presets.editPresetName({
+                                        title = _("Enter new preset name"),
+                                        initial_value = preset_name,
+                                        confirm_button_text = _("Rename"),
+                                    }, function(new_name, dialog_instance)
+                                        if new_name == preset_name then
+                                            UIManager:close(dialog_instance) -- no change?, just close then
+                                            return
+                                        end
+                                        if Presets.validateAndSavePreset(new_name, preset_obj, presets[preset_name]) then
+                                            presets[preset_name] = nil
```

Comment:
How we usually rename a key in a table?
```Lua
t[new_key] = t[old_key]
t[old_key] = nil
```
You break this two connected lines, hiding the saving into other function.
More calls, broken logic. Unnecessary passing of arguments (like `dialog_instance`).
(Name duplication check should be separated from the caller callback)

Reply:
It's so simple:
https://github.com/koreader/koreader/blob/4dc4339bc7e4c767954f4ebb70a00ea33b52de18/frontend/apps/filemanager/filemanagercollection.lua#L1229-L1241
https://github.com/koreader/koreader/blob/4dc4339bc7e4c767954f4ebb70a00ea33b52de18/frontend/apps/filemanager/filemanagercollection.lua#L1271-L1280
(I just rename the menu items. But the preset list is sorted so the menu should be rebuilt, yes)